ruby/compress-image/index.rb

```diff
@@ -0,0 +1,107 @@
+require "json"
+require 'open-uri'
+require "base64"
+require "tinify"
+require "kraken-io"
+
+class NotImageException < StandardError; end
+class ApiError < StandardError; end
+
+invalidPayload = {
+  success: false, message: "Invalid Payload"
+}
+notImage = {
+  success: false, message: "Input file is not an image."
+}
+invalidApi = {
+  success: false, message: "Invalid API."
+}
+
+def tinypng(req, image) 
+    # Get api key
+    if !req.env['TINYPNG_API']
+      raise ApiError
+    end
+
+    Tinify.key = req.env['TINYPNG_API']
+    buffer = Base64.decode64(image)
+    result = Tinify.from_buffer(buffer).to_buffer
+    
+    Base64.encode64(result)
+end
+
+def krakenio(req, image) 
+    if !req.env['KRAKENIO_KEY'] or !req.env['KRAKENIO_SECRET']
+        raise ApiError
+    end
+
+    buffer = Base64.decode64(image)
+    File.open('/file.png', 'wb') do |f|
```

Comment:
Wouldn't writing to `/file.png` cause problems if multiple function executions occur at the same time?

Reply:
Do you have any suggestion on this? I can't find a way to inject the image into the api without having the write to file first.

Reply:
> Do you have any suggestion on this? I can't find a way to inject the image into the api without having the write to file first.

@vzsky writing to a file is fine, but it would be good to write into a random temporary folder per request to make sure there's no clash.

Reply:
Looks like ruby has this: https://ruby-doc.org/stdlib-2.5.1/libdoc/tmpdir/rdoc/Dir.html

Reply:
Oh, ok. will make a commit